modules/social_features/social_post/js/keycode-submit.js

```diff
@@ -10,29 +10,24 @@
   Drupal.behaviors.keycodeSubmit = {
     attach: function (context, settings) {
 
-      // Enable CTRL+Enter to submit a form.
-      var keySubmit = function (form, textarea, submit) {
-        textarea.on("keydown", function (e) {
-          // Make sure the textarea contains text.
-          if ($.trim(textarea.val()) != "") {
-            if ((e.keyCode == 13 && e.ctrlKey) || (e.keyCode == 13 && e.metaKey)){
-              e.preventDefault();
-              submit.prop('disabled', true);
-              form.submit();
-            }
-          }
-        });
-      }
-
       // Post form.
-      var postForm = $('#social-post-entity-form');
-      keySubmit($(postForm), $('.form-textarea', postForm), $('.form-submit', postForm));
-
-      // Comment forms.
-      $('.comment-form').each(function () {
-        keySubmit($(this), $('.form-textarea', this), $('.form-submit', this));
-      });
-
+        $('#social-post-entity-form .form-textarea')
+      .off('keydown', onTextAreaKeyDown)
+      .on('keydown', onTextAreaKeyDown);
+      // comment form
+        $('.comment-form .form-textarea').on('keydown')
+      .off('keydown', onTextAreaKeyDown)
+      .on('keydown', onTextAreaKeyDown);
     }
   };
+  function onTextAreaKeyDown(e) {
+    var textarea = $(this);
+    var submit = textarea.closest('form').find('.form-submit');
+    if ($.trim(textarea.val()) != '') {
+      if ((e.keyCode === 13 && e.ctrlKey) || (e.keyCode === 13 && e.metaKey)) {
```

Comment:
Perhaps we can move this if statement to the top of this function, that way we do the quickest checks first (checking properties of the event object). This way we can avoid doing more intensive checks on each keystroke (DOM lookup and jQuery object creation). This keeps the post field more responsive.